libgalois/include/katana/gstl.h

```diff
@@ -40,51 +40,134 @@ namespace katana {
 
 namespace gstl {
 
-//! [define Pow2VarSizeAlloc]
+//! [STL vector using katana Pow2BlockAllocator]
+//! specializes std::vector to use katana concurrent, scaleable allocator:
+//! the allocator is composed of thread-local allocators
```

Comment:
Suggesting slight change for clarity (correct me if I'm wrong as I haven't looked at that code lately). 

the allocator is composed of thread-local allocators that allocate memory in power-of-2 sized blocks. Allocators use Huge pages if available [See https://www.kernel.org/doc/Documentation/vm/hugetlbpage.txt], and fall back to malloc for larger sizes, which requires a global lock.

Reply:
They always get allocation in terms of pages and they require a global lock to get pages - huge or otherwise, right?

Reply:
> > > Side question: Why is NUMAArray a type? Shouldn't it be an allocator that can be used with any number of other types?
> > 
> > 
> > @arthurp : Asking the right questions. We should have a DynamicArray type, which can use a standard allocator or a Numa Allocator. Then NumArray = DynamicArray + NumaAllocator.
> 
> Agree with Amber except that I'd prefer to call it an `Array` type.
> However, the NumaAllocator is NOT a stand-in replacement for other allocators because it uses all threads to the NUMA allocation. So I'd prefer the current special type instead of an `Array` or `DynamicArray` type whose behavior would change depending on whether it is using a NumaAllocator or not.

That is a good point. I didn't think of that.